### PR TITLE
Marcus/LessonCommand

### PR DESCRIPTION
**Pull request by MarcTzh**
yet to add test cases and documentation

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/LessonCommand.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ArrayList<Task> tasksToAdd = lesson.createRecurringTasks();
+        for (Task taskToAdd: tasksToAdd) {
+            System.out.println("date: " + taskToAdd.getDateTime());
+            if (model.hasTask(taskToAdd)) {
+                System.out.println("B");
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/logic/commands/LessonCommand.java:60]: What are the B, C and D for?

File: src/main/java/seedu/address/logic/parser/LessonCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+        if (argMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
+        }
+        if (argMultimap.getValue(PREFIX_DAY).isPresent()) {
+            dayOfWeek = ParserUtil.parseDay(argMultimap.getValue(PREFIX_DAY).get());
+        } else {
+            throw new ParseException(DateTime.DAY_MESSAGE_CONSTRAINTS);
+        }
+        requireAllNonNull(startDate, endDate, startTime, endTime, dayOfWeek);
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/logic/parser/LessonCommandParser.java:74]: Will there be any case where a valid user input will make any of these null?

File: src/main/java/seedu/address/model/lesson/Lesson.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.Objects;
+
+import seedu.address.model.task.DateTime;
+import seedu.address.model.task.Description;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.Title;
+import seedu.address.model.task.Type;
+
+/**
+ * Lesson class to store information about a module's lessons
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/lesson/Lesson.java:21]: Missing period at the end.

File: src/main/java/seedu/address/model/lesson/Lesson.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    public ArrayList<Task> createRecurringTasks() {
+        LocalDate currentDate = getStartDate();
+        while (!(currentDate.getDayOfWeek()).equals(this.dayOfWeek)) {
+            currentDate = currentDate.plusDays(1);
+        }
+        ArrayList<Task> tasksToAdd = new ArrayList<>();
+        while (currentDate.isBefore(this.endDate) || currentDate.isEqual(this.endDate)) {
+            LocalDateTime localDateTime = LocalDateTime.of(currentDate, getStartTime());
+            //todo:there might be a better way to do this
+            DateTimeFormatter format = DateTimeFormatter.ofPattern("dd-MM-yyyy hh:mm");
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/lesson/Lesson.java:81]: I think you can use the formatter from the DateTime class if possible

File: src/main/java/seedu/address/model/lesson/Lesson.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    /**
+     * Returns true if both lessons of the same title have the same start and end date, and the same start and end time.
+     */
+    public boolean isSameLesson(Lesson otherLesson) {
+        if (otherLesson == this) {
+            return true;
+        }
+        return otherLesson != null
+                && otherLesson.getTitle().equals(getTitle())
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/lesson/Lesson.java:98]: No need day of week?

**Review by BobbyZhouZijian — COMMENTED** [review]
Haven't manually tested. These are just some of my concerns/things that I am a bit confused.

File: src/main/java/seedu/address/logic/commands/LessonCommand.java (hunk 1 above)
Comment (MarcTzh) [reply src/main/java/seedu/address/logic/commands/LessonCommand.java:60]: LOL i thought i deleted, it was for debugging :)

File: src/main/java/seedu/address/model/lesson/Lesson.java (hunk 5 above)
Comment (MarcTzh) [reply src/main/java/seedu/address/model/lesson/Lesson.java:98]: yes need, good spot

**Pull request closed without merging**